src/buckets/components/createBucketForm/UpdateBucketOverlay.tsx

```diff
@@ -131,10 +156,58 @@ const UpdateBucketOverlay: FunctionComponent<Props> = ({
     }
   }
 
+  const isValid = () => {
+    // are there measurement schemas?
+    const haveSchemas =
+      !!newMeasurementSchemaRequests && !!newMeasurementSchemaRequests.length
+
+    if (!haveSchemas) {
+      // no schemas, nothing to validate, so everything is fine
+      return true
+    }
+    // if so, are they all valid?
+    const reducer = (previousValue, currentValue) =>
+      previousValue && currentValue.valid
+
+    const alltrue = newMeasurementSchemaRequests.reduce(reducer, true)
+
+    if (alltrue) {
+      return true
+    }
+    // not all true :(
+
+    // if not valid, decorate them to show they are not valid!
+    setShowSchemaValidation(true)
+    return false
+  }
+
   const handleSubmit = (e: FormEvent<HTMLFormElement>): void => {
     e.preventDefault()
-    onUpdateBucket(bucketDraft)
-    handleClose()
+
+    if (isValid()) {
+      onUpdateBucket(bucketDraft)
+
+      if (newMeasurementSchemaRequests?.length) {
+        const mSchemas = newMeasurementSchemaRequests.map(item => ({
+          columns: item.columns,
+          name: item.name,
+        }))
+
+        // create the measurement schemas one by one here!
+
+        mSchemas.forEach(createRequest => {
+          event('bucket.schema.explicit.editing.uploadSchema')
+          onAddMeasurementSchemaToBucket(
+            bucketDraft.id,
+            bucketDraft.orgID,
+            bucketDraft.name,
+            createRequest
+          )
+        })
```

Comment:
this looks to be another `n*n` loop that doesn't need to loop twice. why not just call `forEach` on `newMeasurementSchemaRequests` and skip mapping over the whole array just to loop over it again?

Reply:
it's 2n (n loop twice), but yeah can put it in one loop

Reply:
ha! I actually commented [the same thing before I saw this comment](https://github.com/influxdata/ui/pull/2766#discussion_r721748759)